libraries/freertos_plus/aws/ota/src/aws_iot_ota_agent.c

```diff
@@ -861,13 +864,19 @@ static OTA_Err_t prvRequestJobHandler( OTA_EventData_t * pxEventData )
 
             /* Send shutdown event to the OTA Agent task. */
             xEventMsg.xEventId = eOTA_AgentEvent_Shutdown;
-            OTA_SignalEvent( &xEventMsg );
 
-            /*
-             * Too many requests have been sent without a response or too many failures
-             * when trying to publish the request message. Abort. Store attempt count in low bits.
-             */
-            xReturn = ( uint32_t ) kOTA_Err_MomentumAbort | ( otaconfigMAX_NUM_REQUEST_MOMENTUM & ( uint32_t ) kOTA_PAL_ErrMask );
+            if( !OTA_SignalEvent( &xEventMsg ) )
```

Comment:
We're changing logic here in addition to fixing MISRA warnings, you may want to make sure that your commit message mentions this.

Reply:
Actually not, I think it's GitHub confusing about the diff, I'm only catching the error returned from `OTA_SignalEvent`.